include/lbann/layers/activations/activation.hpp

```diff
@@ -32,45 +32,32 @@
 
 namespace lbann {
 
+/** Abstract base class for activation layer.
+ *  Activation layers implement the nonlinear activation functions
+ *  common in neural networks.
+ */
 class activation_layer : public Layer {
-
  public:
-  activation_layer(lbann_comm *comm) :
-    Layer(comm) {
-  }
-  activation_layer(const activation_layer&) = default;
-  activation_layer& operator=(const activation_layer&) = default;
-
-  ~activation_layer() override {}
-
-  template<data_layout T_layout> inline void initialize_distributed_matrices() {
-    Layer::initialize_distributed_matrices<T_layout>();
-  }
-
-  virtual void initialize_distributed_matrices() = 0;
-
+  activation_layer(lbann_comm *comm) : Layer(comm) {}
 };
 
+/** Abstract base class for entry-wise activation layer.
+ *  A nonlinear activation function is applied independently to each
+ *  input entry.
+ */
 class entrywise_activation_layer : public activation_layer {
-
  public:
-  entrywise_activation_layer(lbann_comm *comm) :
-    activation_layer(comm) {
-  }
-  entrywise_activation_layer(const entrywise_activation_layer&) = default;
-  entrywise_activation_layer& operator=(
-    const entrywise_activation_layer&) = default;
+  entrywise_activation_layer(lbann_comm *comm) : activation_layer(comm) {}
 
-  ~entrywise_activation_layer() override {}
+ protected:
 
-  template<data_layout T_layout> inline void initialize_distributed_matrices() {
-    activation_layer::initialize_distributed_matrices<T_layout>();
-  }
+  /** Activation function.
+   *  This function is applied independently to each input entry.
+   */
+  virtual DataType activation(DataType x) const = 0;
 
- protected:
-  
-  virtual DataType activation_function(DataType x) = 0;
-  virtual DataType activation_function_gradient(DataType x) = 0;
```

Comment:
This function should be the derivative of a single-variable function, so calling it a "gradient" is less clear.